### PR TITLE
Add defines for all battery backends

### PR DESCRIPTION
**Pull request by peterbarker**
This PR adds `AP_BATTERY_XYZZY_ENABLED` defines for each backend type, and renames several older ones for consistency.

Small changes here and there for the new defines.

### CODE REVIEW

**Comment by peterbarker** [comment]
```
Board              AP_Periph  blimp  bootloader  copter  heli  iofirmware  plane  rover  sub
Durandal                      *      *           *       *                 *      *      *
Hitec-Airspeed     0                                                                     
KakuteH7-bdshot               *      *           *       *                 *      *      *
MatekF405                     0      *           0       0                 0      0      0
Pixhawk1-1M                   -8     *           -8      0                 0      0      0
f103-QiotekPeriph  *                                                                     
f303-Universal     *                                                                     
iomcu                                                          *                         
revo-mini                     *      *           *       *                 *      *      *
skyviper-v2450                                   -2432                                   
```

**Review by tridge — APPROVED** [review]
good to go in with the _config change (or can be a followup PR)

File: libraries/AP_BattMonitor/AP_BattMonitor_Analog.cpp [hunk 1]
```diff
@@ -1,7 +1,10 @@
+#include "AP_BattMonitor_Analog.h"
```
[the remark this answers is not in the document]
Comment (peterbarker) [reply libraries/AP_BattMonitor/AP_BattMonitor_Analog.cpp]: Done

**Comment by peterbarker** [comment]
Ran this through test_build_options.py and size_compare_branches.py and I'm happy with the results.

**Pull request merged**